packages/rocketchat-lib/server/functions/sendMessage.coffee

```diff
@@ -79,6 +79,13 @@ RocketChat.sendMessage = (user, message, room, options) ->
 						query:
 							userId: userOfMention._id
 
+				if userOfMention.status is 'offline' and userOfMention.settings?.preferences?.emailNotificationMode is true
+					Email.send
+						to: userOfMention.emails[0].address
```

Comment:
Do you know off-hand if `userOfMention.emails` is `undefined` or `[]` if a user has no emails? I'll check it in the if statement.